### PR TITLE
Switch to zserio 2.4.2

### PR DESCRIPTION
**Pull request by l-laura**
Since current services are mainly zserio 2.4.2 compatible, I changed zsomeip to use it by default. The flag is documented in README and I tested the weather service using both zserio versions.

### CODE REVIEW

File: client-service/ZsomeIpService.cpp [hunk 1]
```diff
@@ -72,47 +78,87 @@ ZsomeIpClient::ZsomeIpClient(
...
+        std::unique_lock<std::mutex> lock_until_response(running_mutex_);
+        app_->send(request);
+        response_arrived_.wait(lock_until_response);
+
+        if (response_code_ != vsomeip::return_code_e::E_OK) {
+            throw ZsomeIpRuntimeError(response_code_);
+        }
+
+        auto* payload = static_cast<uint8_t*>(response_payload_->get_data());
+        uint32_t length = response_payload_->get_length();
```
Comment (josephbirkner) [inline client-service/ZsomeIpService.cpp:155]: Consider `auto`? (also above?)

File: client-service/ZsomeIpService.h [hunk 2]
```diff
@@ -33,15 +34,29 @@ class ZsomeIpService : public ZsomeIpApp {
...
-    ZsomeIpClient(
-        const std::string& appName,
-        std::shared_ptr<MethodDefinition> methodDefinition);
-    std::vector<uint8_t> callMethod(
-        zserio::StringView methodName,
-        const zserio::IServiceData& requestData,
-        void* context) override;
+class ZsomeIpClient
+#ifdef ZSERIO_2_4_2_SERVICE_INTERFACE
+            : public ZsomeIpApp, public zserio::IService {
```
Comment (josephbirkner) [inline client-service/ZsomeIpService.h:39]: Consider to align indentation usages of `#if` and `#else` blocks

**Review by josephbirkner — CHANGES_REQUESTED** [review]
Looks good, but I have a small change request for how a user can switch the used version...

File: CMakeLists.txt [hunk 3]
```diff
@@ -23,6 +23,11 @@ endif()
 
 add_subdirectory(definitions)
 
+################################################################
+# Remove this definition to use zserio 2.5.0 service interface #
+################################################################
+add_compile_definitions(ZSERIO_2_4_2_SERVICE_INTERFACE)
```
[the remark this answers is not in the document]
Comment (l-laura) [reply CMakeLists.txt]: That's a good idea - the change is in, although I renamed the option to ``ZSOMEIP_USE_2_5_0_ZSERIO_SERVICE_INTERFACE`` for clarity. Also updated documentation on the option.

**Pull request merged**